### PR TITLE
Factor out subcomponents of info panel

### PR DESCRIPTION
**Pull request by akalin-keybase**
This is in preparation for merging all the info panels into one component
that pushes sub-components onto a virtualized list.

Also move the mute row into the Notifications component and make it always
have the same label, per the design. This is also in preparation for making
a checked mute box disable/hide the other notification components.

### CODE REVIEW

File: shared/chat/conversation/info-panel/header.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import {Avatar, Box, ClickableBox, Text} from '../../../common-adapters'
+import {globalMargins, globalStyles, isMobile} from '../../../styles'
+
+type SmallProps = {
+  teamname: string,
+  participantCount: number,
+  onClick: () => void,
+}
+
+const SmallTeamHeader = ({teamname, participantCount, onClick}: SmallProps) => (
```
Comment (akalin-keybase) [inline shared/chat/conversation/info-panel/header.js]: No new code here, just moved from index.js

File: shared/chat/conversation/info-panel/notifications/container.js [hunk 2]
```diff
@@ -98,6 +108,5 @@ export default compose(
       this.props._resetNotificationSaveState()
     },
   }),
-  // $FlowIssue doesn't like dynamic props like we do above
-  branch(props => !props.conversationIDKey, renderNothing)
+  branch(props => !('conversationIDKey' in props), renderNothing)
```
Comment (buoyad) [inline shared/chat/conversation/info-panel/notifications/container.js]: ah I've never seen this syntax before. the change fixes the `$FlowIssue` I'm guessing?

Comment (chrisnojima) [reply shared/chat/conversation/info-panel/notifications/container.js]: we shouldn't be using branch at all for these types of things

Comment (buoyad) [reply shared/chat/conversation/info-panel/notifications/container.js]: it also looks like `'conversationIDKey' in props` will be true if  
```js
props = {conversationIDKey: undefined}
```
which might be bad

Comment (akalin-keybase) [reply shared/chat/conversation/info-panel/notifications/container.js]: Reverted this. Agree that we want to get rid of branch entirely, but I didn't want to make this PR bigger. Will do in a separate PR.

**Pull request merged**